syft/messaging/message.py

```diff
@@ -240,11 +238,18 @@ def unbufferize(worker: AbstractWorker, protobuf_obj: "CommandMessagePB") -> "Co
         Examples:
             message = unbufferize(sy.local_worker, protobuf_msg)
         """
-        detailed = ComputationAction.unbufferize(worker, protobuf_obj.action)
-
-        return CommandMessage(
-            detailed.name, detailed.target, detailed.args, detailed.kwargs, detailed.return_ids
-        )
+        if protobuf_obj.computation.ByteSize():
+            detailed = ComputationAction.unbufferize(worker, protobuf_obj.computation)
+            action = ComputationAction(
+                detailed.name, detailed.target, detailed.args, detailed.kwargs, detailed.return_ids
+            )
+        elif protobuf_obj.communication.ByteSize():
+            detailed = CommunicationAction.unbufferize(worker, protobuf_obj.communication)
```

Comment:
Once you retrieve the `action` from the Protobuf object, you should be able to call `sy.serde.protobuf.serde._unbufferize` on it and pass the result into the `CommandMessage` constructor below.

Reply:
Oh, didn't see this message before answering to the previous one...

Reply:
Ha, same! I didn't read far enough to have the whole idea in mind yet when I wrote the comment above.